lib/pbench/test/unit/agent/test_tool_group.py

```diff
@@ -268,3 +270,52 @@ def mock_read_text(path):
                 assert tg.get_label(host) == labels[host]
             assert tg.get_tools("bad-host") == {}
             assert tg.get_label("bad-host") == ""
+
+
+def test_gen_tool_groups():
+    """Verify the tool group directory generator."""
+
+    class MockPath:
+        """A simple ToolGroup mock to generate host names from different
+        tool group directories.
+        """
+
+        _groups = set(("heavy", "light", "medium"))
+
+        def __init__(self, name: str):
+            self._name = name
+
+        def glob(self, prefix: str):
+            if self._name == "no-tool-group-dirs":
+                # The special name which for a directory which "does not
+                # have tool groups".
+                return
+            for n in self._groups:
+                # Return the 3 tools groups that are found with the prefix
+                # attached.
+                yield Path(f"{prefix[:-2]}-{n}")
```

Comment:
Why does this remove the last character (i.e., the `-`) and then add a `-`?  Why doesn't it just
```
                yield Path(prefix + n)
```

Reply:
Because in this case it is accepting the values passed to it from `get_tool_groups`, which is `<something>-v1-*`, and we want to return `<something>-v1-heavy`, etc.

Reply:
Sorry, I was "off by one".  How about?
```
                yield Path(prefix[:-1] + n)
```
😁